src/hooks/useKeyboardNavigation.js

```diff
@@ -27,11 +29,68 @@ const useKeyboardNavigation = ({
 
   useEffect(() => {
     return tinykeys(emojiListRef.current, {
-      ArrowRight: focusNextEmoji,
-      ArrowLeft: focusPrevEmoji,
+      ArrowRight: () => navigateGrid(RIGHT),
+      ArrowLeft: () => navigateGrid(LEFT),
+      ArrowUp: () => navigateGrid(UP),
+      ArrowDown: () => navigateGrid(DOWN),
     });
   }, []);
 
+  const updateActiveItem = newActiveItem => {
+    if (newActiveItem && newActiveItem.firstChild)
+      newActiveItem.firstChild.focus();
+  };
+
+  const navigateGrid = direction => {
+    const grid = getCurrentEmojiListGroup();
+    const active = getActiveElement().parentElement;
+    const gridChildren = Array.from(grid.children);
+    const activeIndex = gridChildren.indexOf(active);
+
+    const numOfItems = gridChildren.length;
+    const baseOffset = gridChildren[0].offsetTop;
+    const breakIndex = gridChildren.findIndex(
+      item => item.offsetTop > baseOffset
+    );
+    const itemsPerRow = breakIndex === -1 ? numOfItems : breakIndex;
+
+    const isTopRow = activeIndex <= itemsPerRow - 1;
+    const isBottomRow = activeIndex >= numOfItems - itemsPerRow;
+    const isLeftColumn = activeIndex % itemsPerRow === 0;
+    const isRightColumn =
+      activeIndex % itemsPerRow === itemsPerRow - 1 ||
+      activeIndex === numOfItems - 1;
+
+    const currentColumn = activeIndex % itemsPerRow;
+
+    const isLastRow =
+      activeIndex >= numOfItems - (numOfItems % itemsPerRow) ||
+      numOfItems % itemsPerRow === 0;
```

Comment:
While I don't have anything specifically against it (other than the fact that we need to run it through some edge cases to see that we're not missing anything), you noticed yourself that these are unrelated to one another, and you broken them down to separate blocks.

It would make sense to write separate functions to handle the different aspects of grid navigation logic. Each part would be easier to understand on its own, and you would only share the things you actually use. For example `baseOffset` is only used once.